db/dump.py

```diff
@@ -112,45 +186,70 @@ def dump_db(location, threads=None, incremental=False, dump_id=None):
-
-            archive_tables_dir = os.path.join(temp_dir, "abdump", "abdump")
-            utils.path.create_path(archive_tables_dir)
-            _copy_tables(archive_tables_dir, start_t, end_t)
-            tar.add(archive_tables_dir, arcname=os.path.join(archive_name, "abdump"))
+    _dump_tables(
+        archive_path=archive_path,
+        threads=threads,
+        dataset_dump=False,
+        time_now=time_now,
+        start_t=start_t,
+        end_t=end_t,
+    )
+    return archive_path
 
-            shutil.rmtree(temp_dir)  # Cleanup
 
-        pxz.stdin.close()
+def _copy_table_into_multiple_files(cursor, table_name, query, tar, archive_name):
+    """Copies data from a table into multiple files and add them to the archive
 
-    return archive_path
+    Args:
+        cursor: a psycopg2 cursor
+        table_name: the name of the table to be copied.
+        query: the select query for getting data from the table, with appropriate LIMIT and OFFSET parameters.
+        tar: the TarFile object to which the table dumps must be added
+        archive_name: the name of the archive
+    """
+    location = tempfile.mkdtemp()
+    offset = 0
+    file_count = 0
+    utils.path.create_path(os.path.join(location, table_name))
+    while True:
+        more_rows_added = False
+        file_count += 1
+        file_name = '{table_name}-{file_number}'.format(table_name=table_name, file_number=file_count)
+        path = os.path.join(location, table_name, file_name)
+        with open(path, "a") as f:
+            logging.info(" - Copying table {table_name} to {file_name}...".format(table_name=table_name, file_name=file_name))
+            current_query = query.format(limit=ROWS_PER_FILE, offset=offset)
+            copy_query = 'COPY ({query}) TO STDOUT'.format(query=current_query)
+            cursor.copy_expert(copy_query, f)
+            offset += ROWS_PER_FILE
+            if f.tell() > 0:
+                more_rows_added = True
+        if more_rows_added:
+            tar.add(path,
+                    arcname=os.path.join(archive_name, "abdump", table_name, file_name))
+
+        os.remove(path)
+        if not more_rows_added:
+            break
+    shutil.rmtree(location)
+
+
+def _copy_table(cursor, location, table_name, query):
+    """Copies data from a table into a file within a specified location.
+    Args:
+        cursor: a psycopg2 cursor
+        location: the directory where the table should be copied.
+        table_name: the name of the table to be copied.
+        query: the select query for getting data from the table.
+    """
+    with open(os.path.join(location, table_name), "w") as f:
+        logging.info(" - Copying table {table_name}...".format(table_name=table_name))
+        print(" - Copying table {table_name}...".format(table_name=table_name))
+        copy_query = 'COPY ({query}) TO STDOUT'.format(query=query)
+        cursor.copy_expert(copy_query, f)
```

Comment:
I think it was undocumented behaviour. 

copy_to constructs the copy query here and it could possibly work with the queries we passed. https://github.com/psycopg/psycopg2/blob/153b0983c18cf6f5536c4d276af2b63d8a5f0568/psycopg/cursor_type.c#L1524